tests/plugins/career/test_routes.py

```
@@ -174,3 +174,131 @@ def test_double_download_409s(client, monkeypatch):
+
+def test_prepare_rejects_a_non_list_songs_value(tmp_path, meta_db, client):
+    # A str is iterable: without the list check, "abc" would prepare three
+    # one-character "songs".
+    for bad in ("abc", 42, {"a": 1}, None):
+        res = client.post("/api/plugins/career/gigs/prepare", json={"songs": bad})
+        assert res.status_code == 200
+        assert res.json()["prepared"] == 0
+
+
+def test_prepare_ignores_non_string_and_blank_entries(tmp_path, meta_db):
+    dlc = tmp_path / "dlc"; dlc.mkdir()
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)